modules/renter/hostdb/scan.go

```diff
@@ -69,12 +73,18 @@ func (hdb *HostDB) queueScan(entry modules.HostDBEntry) {
 			if exists {
 				entry = recentEntry
 			}
+
+			// Create new worker thread
+			if hdb.scanningThreads < maxScanningThreads {
+				hdb.scanningThreads++
+				go hdb.threadedProbeHosts(scanPool)
```

Comment:
The increment + decrement should be near each other in the source code. So either add the decrement here, like so:
```go
hdb.scanningThreads++
go func() {
    hdb.threadedProbeHosts(scanPool)
    hdb.mu.Lock()
    hdb.scanningThreads--
    hdb.mu.Unlock()
}()
```
or move the increment into `threadedProbeHosts`. I think the former makes it easier to verify correctness. As written, `scanningThreads` will not be properly decremented if `threadedProbeHosts` returns early.